packages/keychain/src/pages/index.tsx

```diff
@@ -73,7 +85,7 @@ function Home() {
         return <></>;
       }
 
-      if (controller.session(policies)) {
+      if (hasSessionForPolicies) {
```

Comment:
Hmm does this migration to async maintain the same behavior? It seems on initial render it will be false and proceed now. I think we might need a loading state for the intermediate state, when the `useEffect` hasn't been executed

Reply:
I think you're right that this is _technically_ not the exact same behaviour as the sync version. However, I think under a real world scenario it's _very_ unlikely that users would ever run into the case where the loading state is observable. This is because when `.connect()` is called, it's highly unlikely that the host app would somehow also be blocking the wasm module. After all, `.connect()` is likely their first function call before doing anything real. This means that despite being `async`, the `hasSessionForPolicies` state would instantaneously be set to its desired value.

That said, I can certainly add a loading state. It's just that given the unlikeliness of it being actually observable as discussed above, I'd go for something very primitive. I will test it by artificially delaying the wasm call.

Reply:
Alright I just pushed a commit adding a loading screen using the existing `<PageLoading />` component:

<img src="https://github.com/user-attachments/assets/1ccf25c6-3d09-48cd-ac44-7a5727c75a09" width="300">